regex/tests/Test.lean

```diff
@@ -153,4 +153,907 @@ def word_class := Regex.parse! r##"\w+"##
+
+-- name = "wb34"
+#guard word_boundary_18.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard word_boundary_18.bt.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+def word_boundary_19 := Regex.parse! r##"^(?:\b).(?:\b)$"##
+
+-- name = "wb35"
+#guard word_boundary_19.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard word_boundary_19.bt.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+def word_boundary_20 := Regex.parse! r##"^^^^^\b$$$$$"##
+
+-- name = "wb36"
+#guard word_boundary_20.captureAll "" = #[]
+#guard word_boundary_20.bt.captureAll "" = #[]
+
+def word_boundary_21 := Regex.parse! r##"^^^^^(?:\b).$$$$$"##
+
+-- name = "wb37"
+#guard word_boundary_21.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard word_boundary_21.bt.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+def word_boundary_22 := Regex.parse! r##"^^^^^\b$$$$$"##
+
+-- name = "wb38"
+#guard word_boundary_22.captureAll "x" = #[]
+#guard word_boundary_22.bt.captureAll "x" = #[]
+
+def word_boundary_23 := Regex.parse! r##"^^^^^(?:\b\b\b).(?:\b\b\b)$$$$$"##
+
+-- name = "wb39"
+#guard word_boundary_23.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard word_boundary_23.bt.captureAll "x" = #[
+  ⟨"x", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+def word_boundary_24 := Regex.parse! r##"(?:\b).+(?:\b)"##
+
+-- name = "wb40"
+#guard word_boundary_24.captureAll "$$abc$$" = #[
+  ⟨"$$abc$$", #[.some ⟨2⟩, .some ⟨5⟩]⟩
+]
+#guard word_boundary_24.bt.captureAll "$$abc$$" = #[
+  ⟨"$$abc$$", #[.some ⟨2⟩, .some ⟨5⟩]⟩
+]
+
+def word_boundary_25 := Regex.parse! r##"\b"##
+
+-- name = "wb41"
+#guard word_boundary_25.captureAll "a b c" = #[
+  ⟨"a b c", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+  ⟨"a b c", #[.some ⟨1⟩, .some ⟨1⟩]⟩,
+  ⟨"a b c", #[.some ⟨2⟩, .some ⟨2⟩]⟩,
+  ⟨"a b c", #[.some ⟨3⟩, .some ⟨3⟩]⟩,
+  ⟨"a b c", #[.some ⟨4⟩, .some ⟨4⟩]⟩,
+  ⟨"a b c", #[.some ⟨5⟩, .some ⟨5⟩]⟩
+]
+#guard word_boundary_25.bt.captureAll "a b c" = #[
+  ⟨"a b c", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+  ⟨"a b c", #[.some ⟨1⟩, .some ⟨1⟩]⟩,
+  ⟨"a b c", #[.some ⟨2⟩, .some ⟨2⟩]⟩,
+  ⟨"a b c", #[.some ⟨3⟩, .some ⟨3⟩]⟩,
+  ⟨"a b c", #[.some ⟨4⟩, .some ⟨4⟩]⟩,
+  ⟨"a b c", #[.some ⟨5⟩, .some ⟨5⟩]⟩
+]
+
+def word_boundary_26 := Regex.parse! r##"\bfoo\b"##
+
+-- name = "wb42"
+#guard word_boundary_26.captureAll "zzz foo zzz" = #[
+  ⟨"zzz foo zzz", #[.some ⟨4⟩, .some ⟨7⟩]⟩
+]
+#guard word_boundary_26.bt.captureAll "zzz foo zzz" = #[
+  ⟨"zzz foo zzz", #[.some ⟨4⟩, .some ⟨7⟩]⟩
+]
+
+def word_boundary_27 := Regex.parse! r##"\b^"##
+
+-- name = "wb43"
+#guard word_boundary_27.captureAll "ab" = #[
+  ⟨"ab", #[.some ⟨0⟩, .some ⟨0⟩]⟩
+]
+#guard word_boundary_27.bt.captureAll "ab" = #[
+  ⟨"ab", #[.some ⟨0⟩, .some ⟨0⟩]⟩
+]
+
+-- Non word boundary tests
+-- Tests for \B. Note that \B is not allowed if UTF-8 mode is enabled, so we
+-- have to disable it for most of these tests. This is because \B can match at
+-- non-UTF-8 boundaries.
```

Comment:
```suggestion
```

Since the intended behavior of this implementation is always enable Unicode mode, let's remove this comment.